realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -100,18 +100,40 @@ public String getClassName() {
     }
 
     /**
-     * Sets a new name for this RealmObject class. This is equivalent to renaming it.
+     * Sets a new name for this RealmObject class. This is equivalent to renaming it. When {@link RealmObjectSchema#table}
+     * has a primary key, this will transfer the primary key for the new class name.
      *
      * @param className the new name for this class.
+     * @throws IllegalArgumentException if {@param className} is {@code null} or an empty string, or its length exceeds 57 characters.
      * @see RealmSchema#rename(String, String)
      */
     public RealmObjectSchema setClassName(String className) {
         checkEmpty(className);
+        if (57 < className.length()) {
```

Comment:
Please avoid magic constants in code. Perhaps extract this into a named constant like perhaps `CLASSNAME_MAX_LENGTH = 64 - TABLE_PREFIX.length`
or something like that.